src/main.rs

```diff
@@ -1,18 +1,37 @@
+use std::env;
+use std::process::exit;
+
+use clap::{crate_authors, Clap};
+
 mod rusty_hook;
 
-use std::process::exit;
+#[derive(Clap)]
+#[clap(version = env!("CARGO_PKG_VERSION"), author = crate_authors!())]
+struct RustyHookOpts {
+    #[clap(subcommand)]
+    subcmd: RustyHookSubCommand,
+}
 
-extern crate ci_info;
-extern crate getopts;
-extern crate nias;
-use getopts::Options;
-use std::env;
+#[derive(Clap)]
+enum RustyHookSubCommand {
+    /// Initialize rusty-hook's git hooks in the current directory.
+    #[clap(version = env!("CARGO_PKG_VERSION"), author = crate_authors!())]
+    Init,
+    /// Run a git hook using the current directory's configuration.
+    /// Ran automatically by rusty-hook's git hooks.
+    #[clap(version = env!("CARGO_PKG_VERSION"), author = crate_authors!())]
+    Run(RustyHookRun),
+}
 
-fn print_version() {
-    println!(env!("CARGO_PKG_VERSION"));
+#[derive(Clap)]
+struct RustyHookRun {
+    #[clap(long)]
+    hook: String,
+    #[clap(name = "git args", raw(true))]
+    args: Vec<String>,
```

Comment:
Ultimately for the git args we'll need a single string on the rust side that can be used to set the value for an env variable for the context the command is running, so that they can easily be made available for the hook commands/scripts. Could we get clap to just condense this to a single String? I'd like to avoid having to deal with the arg vs flag/opt interpretation and mapping to multiple env variables

```toml
commit-msg = "foo $RUSTY_HOOK_GIT_PARAMS"

Reply:
could just `args.join(" ")`, more flexible to bring it in as a Vec

Reply:
> more flexible to bring it in as a Vec

Sure, but I can't really envision where that flexibility would really be needed (famous last words :wink:) since there's just the one value we'll be setting in the env. I suppose it doesn't matter all that much since there won't be a ton of args getting passed through and, should there be a need to change, we could do so without requiring a breaking change to the interface